src/js/webrequest.js

```diff
@@ -855,15 +856,55 @@ function dispatcher(request, sender, sendResponse) {
   }
 }
 
+
+/**
+ * Cancels all requests, populates list of pending tabs to be reloaded later,
+ * when Privacy Badger is ready.
+ *
+ * @param {Object} details chrome.webRequest request details
+ * @returns {Object}
+ */
+function onBeforeReady(details) {
+  if (!pendingTabs || details.tabId < 0) {
```

Comment:
when would pendingTabs be falsey here?

Reply:
We set `pendingTabs` to `null` at the end of `reloadPendingTabs()`. Is it possible to still have requests queued for `onBeforeReady()` after `stopBeforeReadyListener()` is called? I would have to look into it.

Reply:
We may want to ignore "main_frame" resources to avoid running into https://github.com/gorhill/uBlock/issues/3130

Reply:
Should the listener not be removed for it to be treated as "persistent" by Firefox? See https://github.com/uBlockOrigin/uBlock-issues/issues/128#issuecomment-435594531